device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -365,7 +379,33 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
         m_stream.synchronize();
 
         /*****************************************************************
-         * Kernel4: Propagate to the next surface
+         * Kernel4: Get key and value for parameter sorting
+         *****************************************************************/
+
+        vecmem::data::vector_buffer<device::sort_key> keys_buffer(
+            global_counter_host.n_candidates, m_mr.main);
+        vecmem::data::vector_buffer<unsigned int> param_ids_buffer(
+            global_counter_host.n_candidates, m_mr.main);
+
+        nThreads = m_warp_size * 2;
+
+        if (global_counter_host.n_candidates > 0) {
```

Comment:
:thinking: Should we not abort "more explicitly" when we don't have candidates anymore? It seems silly to still call `thrust::sort_by_key` and whatnot on empty containers.

Reply:
I will put the sorting part into the block then